packages/hash/frontend/src/lib/subgraph.ts

```diff
@@ -153,21 +166,56 @@ export const getPersistedEntityType = (
   return vertex.inner;
 };
 
+/**
+ * Gets an `Entity` by its `entityId` from within the vertices of the subgraph. Returns `undefined` if
+ * the entity couldn't be found.
+ *
+ * @param subgraph
+ * @param entityId
+ * @throws if the vertex isn't an `EntityVertex`
+ *
+ * @todo - version is required to identify a specific instance of an entity
+ *   https://app.asana.com/0/1202805690238892/1203214689883091/f
+ */
+export const getEntity = (
+  subgraph: Subgraph,
+  entityId: string,
+): Entity | undefined => {
+  const vertex = subgraph.vertices[entityId];
+
+  if (!vertex) {
+    return undefined;
+  }
+
+  if (!isEntityVertex(vertex)) {
+    throw new Error(`expected entity vertex but got: ${vertex.kind}`);
+  }
+
+  return vertex.inner;
+};
+
 /** @todo - getPersistedEntity and getPersistedLink - https://app.asana.com/0/0/1203157172269853/f */
 
 /**
- * Returns all root vertices of the subgraph
+ * Returns all root `Entity` vertices of the subgraph
  *
  * @param subgraph
+ * @throws if the roots aren't all `Entity`
  */
-export const roots = (subgraph: Subgraph): Vertex[] => {
+export const rootsAsEntities = (subgraph: Subgraph): Entity[] => {
```

Comment:
Nit: this feels like inconsistent naming, shouldn't it be something like
```suggestion
export const getRootsAsEntities = (subgraph: Subgraph): Entity[] => {
```

Reply:
Hm, it does feel a bit inconsistent. I'm not sure about which to go for, the others below for example are:
`persistedLinkTypes`
`persistedEntityTypes`
etc.

I basically wanted it to behave like `subgraph.persistedEntityTypes` (although we have to pass `subgraph` as a param).

But I think `getEntityType` was clearer for the single case.

I'm happy to swap either set over, and maybe we should apply this https://github.com/hashintel/hash/pull/1242#discussion_r1003557107 to this situation

Reply:
> I'm happy to swap either set over, and maybe we should apply this https://github.com/hashintel/hash/pull/1242#discussion_r1003557107 to this situation

This probably applies to this thread as well, yeah.